File: Go/1764_Form_Array_By_Concat.go
func canChoose(groups [][]int, nums: []int) bool {
	j, M := 0, len(nums)

	for _, grp := range groups {
		N := len(grp)
		i, pastJ := 0, j

		for i < N && j < M {
			if grp[i] == nums[j] {
				i += 1
			} else {
				i, j = 0, pastJ
				pastJ += 1
			}
			j += 1
		}
		if i < N{
			return false
		}
	}
	return true;
}

File: JS/1764_Form_Array_By_Concat.js
function canChoose(groups, nums){
  let j = 0, M = nums.length

  for (let grp of groups){
    let N = grp.length
    let i = 0, pastJ = j

    while (i < N && j < M){
      if (grp[i] == nums[j]){
        i += 1
      } else {
        i = 0
        j = pastJ
        pastJ += 1
      }
      j += 1
    }
    if (i < N){
      return false
    }
  }
  return true
}
